docs/extending_wsimod.md

```diff
@@ -0,0 +1,67 @@
+# Extending and customising WSIMOD functionality
+
+WSIMOD has many features built-in, but undoubtedly, to model real life scenarios, you will
+need to customise it to your specific needs.
+
+WSIMOD offers 3 ways of doing this customisation of increased complexity - and flexibility:
+overrides, patches and custom classes.
```

Comment:
Probably also to mention here is custom orchestration? @liuly12 do we have any documentation on it yet?